src/cargo/util/toml/targets.rs

```diff
@@ -149,11 +165,7 @@ fn clean_lib(
                 ));
                 legacy_path
             } else {
-                bail!(
-                    "can't find library `{}`, \
-                     rename file to `src/lib.rs` or specify lib.path",
-                    lib.name()
-                )
+                return bail();
```

Comment:
having to add the `return` here concerns me. is it correct?

Reply:
I would thing that just `bail()` (without both return and semicolon) should work?

Reply:
me too! but looks like it doesn't. I blame macro shenanigans?

```rust
error[E0308]: if and else have incompatible types
   --> src/cargo/util/toml/targets.rs:159:13
    |
159 | /             if legacy_path.exists() {
160 | |                 warnings.push(format!(
161 | |                     "path `{}` was erroneously implicitly accepted for library `{}`,\n\
162 | |                      please rename the file to `src/lib.rs` or set lib.path in Cargo.toml",
...   |
168 | |                 bail()
169 | |             }
    | |_____________^ expected struct `std::path::PathBuf`, found enum `std::result::Result`
    |
    = note: expected type `std::path::PathBuf`
               found type `std::result::Result<std::option::Option<core::manifest::Target>, failure::Error>`
```